dev/make-distribution.sh

```diff
@@ -161,7 +161,7 @@ fi
 # Build uber fat JAR
 cd "$SPARK_HOME"
 
-export MAVEN_OPTS="${MAVEN_OPTS:--Xss128m -Xmx4g -Xmx4g -XX:MaxMetaspaceSize=2g -XX:ReservedCodeCacheSize=128m}"
+export MAVEN_OPTS="${MAVEN_OPTS:--Xss128m -Xmx4g -Xmx4g -XX:ReservedCodeCacheSize=128m}"
```

Comment:
Thanks for the fix. I can reproduce this issue through github action: https://github.com/wangyum/make-distribution.sh/actions/runs/3425964379/jobs/5707275344

Reply:
Thank @wangyum for helping check

Reply:
<img width="1171" alt="image" src="https://user-images.githubusercontent.com/1475305/200793242-a9fabc22-d83c-4a10-9a04-3942ed2762d5.png">

https://github.com/LuciferYang/make-distribution.sh/actions/runs/3426521418

I fork your repo and I modified the script to  executed `gh pr checkout 38577` to switch to this branch, then run successfully

Thanks @wangyum